### PR TITLE
New VG broader group control

### PR DESCRIPTION
**Pull request by v-soukum**
## Description
</br>
New VG control added

ADO_VariableGroup_AuthZ_Restrict_Broader_Group_Access_On_VG_With_Secrets.

## Checklist
- [ ] I have read the instructions mentioned in [Contribute to Code](/Contributing.md).
- [ ] I have read and understood the criteria described under [submitting changes](/Contributing.md#submitting-changes). 
- [ ] The title of the PR clearly describes the intent of the PR.
- [ ] This PR does not introduce any breaking changes to the code.

### CODE REVIEW

**Review by SaikumarRagala — DISMISSED** [review]

File: src/AzSK.ADO/Framework/Configurations/SVT/ADO/ADO.VariableGroup.json [hunk 1]
```diff
@@ -89,6 +89,23 @@
               "MSW"
             ],
             "Enabled": true
+          },
+          {
+            "ControlID": "ADO_VariableGroup_AuthZ_Restrict_Broader_Group_Access_On_VG_With_Secrets",
+            "Description": "Broader groups (contributors, project valid users, etc.) should not have user/administrator privileges on variable group contains secrets.",
+            "Id": "VariableGroup170",
```
Comment (juhi037) [inline src/AzSK.ADO/Framework/Configurations/SVT/ADO/ADO.VariableGroup.json:96]: may be it should be like "...on variable groups containing secrets." or "on variable group which contains secrets."

Comment (v-soukum) [reply src/AzSK.ADO/Framework/Configurations/SVT/ADO/ADO.VariableGroup.json]: changes done

File: src/AzSK.ADO/Framework/Configurations/SVT/ControlSettings.json [hunk 2]
```diff
@@ -468,6 +472,16 @@
         "(?# To match ADO PATs.)^[a-z2-7]{52}$"
       ]
     },
+    {
+      "RegexCode": "SecretsInVariables",
```
Comment (t-rbavdekar) [inline src/AzSK.ADO/Framework/Configurations/SVT/ControlSettings.json:476]: Does not detect pass@1234 but can detect Pass@1234

**Review by t-rbavdekar — APPROVED** [review]

**Review by Garima-msft — APPROVED** [review]

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.VariableGroup.ps1 [hunk 3]
```diff
@@ -327,4 +327,99 @@ class VariableGroup: ADOSVTBase
...
+    hidden [ControlResult] CheckBroaderGroupAccessForVarGrpWithSecrets([ControlResult] $controlResult)
+    {
+        $controlResult.VerificationResult = [VerificationResult]::Failed
+        try 
+        {
+            if ($this.ControlSettings -and [Helpers]::CheckMember($this.ControlSettings, "VariableGroup.RestrictedBroaderGroupsForVariableGroup") -and [Helpers]::CheckMember($this.ControlSettings, "VariableGroup.RestrictedRolesForBroaderGroupsInVariableGroupContainingSecrets"))
+            {
+                $restrictedBroaderGroupsForVarGrp = $this.ControlSettings.VariableGroup.RestrictedBroaderGroupsForVariableGroup;
+                $restrictedRolesForBroaderGroupsInvarGrp = $this.ControlSettings.VariableGroup.RestrictedRolesForBroaderGroupsInVariableGroupContainingSecrets;
+                $patterns = @($this.ControlSettings.Patterns | Where-Object {$_.RegexCode -eq "SecretsInVariables"} | Select-Object -Property RegexList);
```
Comment (t-rbavdekar) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.VariableGroup.ps1:339]: can we load patterns inside next if condition

Comment (v-soukum) [reply src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.VariableGroup.ps1]: done

**Review by t-rbavdekar — APPROVED** [review]

**Review by SaikumarRagala — APPROVED** [review]

**Pull request merged**